core/src/al/miniaudio_context.cpp

```diff
@@ -0,0 +1,350 @@
+            return;
+        }
+
+        ma_engine_config engineConfig = ma_engine_config_init();
+
+        if (listenerCount > MA_ENGINE_MAX_LISTENERS)
+        {
+            CUBOS_FAIL("Maximum number of listeners is 4");
+            return;
+        }
+
+        engineConfig.listenerCount = listenerCount;
+        engineConfig.pPlaybackDeviceID = deviceId; // Use the found device ID
+
+        if (ma_engine_init(&engineConfig, &mEngine) != MA_SUCCESS)
+        {
+            CUBOS_FAIL("Failed to initialize audio engine");
+            return;
+        }
+
+        mListeners.reserve(listenerCount);
+        for (ma_uint32 i = 0; i < listenerCount; ++i)
+        {
+            mListeners.emplace_back(std::make_shared<MiniaudioListener>(mEngine, i));
+        }
+    }
+
+    ~MiniaudioDevice() override
+    {
+        ma_device_uninit(&mDevice);
+    }
+
+    Source createSource() override
+    {
+        return std::make_shared<MiniaudioSource>(mEngine);
+    }
+
+    Listener listener(size_t index) override
+    {
+        if (index >= mListeners.size())
+        {
+            CUBOS_ERROR("Listener index out of range");
+            return nullptr;
+        }
+        return mListeners[index];
+    }
+
+private:
+    ma_context mContext;
+    ma_device mDevice;
+    ma_engine mEngine;
+    std::vector<std::shared_ptr<MiniaudioListener>> mListeners;
+};
+
+MiniaudioContext::MiniaudioContext()
+{
+    if (ma_context_init(nullptr, 0, nullptr, &mContext) != MA_SUCCESS)
+    {
+        CUBOS_FAIL("Failed to initialize audio context.");
+        return;
+    }
+}
+
+MiniaudioContext::~MiniaudioContext()
+{
+    ma_context_uninit(&mContext);
+}
+
+std::string MiniaudioContext::getDefaultDevice()
+{
+    // Create a temporary context
+    ma_context context;
+    if (ma_context_init(nullptr, 0, nullptr, &context) != MA_SUCCESS)
+    {
+        CUBOS_ERROR("Failed to initialize audio context for default device lookup");
+        return "";
+    }
+
+    ma_device_info* pPlaybackDeviceInfos;
+    ma_uint32 playbackDeviceCount;
+
+    if (ma_context_get_devices(&context, &pPlaybackDeviceInfos, &playbackDeviceCount, nullptr, nullptr) != MA_SUCCESS)
+    {
+        CUBOS_ERROR("Failed to enumerate audio devices when searching for default");
+        return "";
+    }
+
+    ma_context_uninit(&context);
+
+    for (ma_uint32 i = 0; i < playbackDeviceCount; i++)
+    {
+        if (pPlaybackDeviceInfos[i].isDefault)
+        {
+            return pPlaybackDeviceInfos[i].name;
+        }
+    }
+
+    CUBOS_WARN("No default audio device found");
+    return "";
+}
+
+void MiniaudioContext::enumerateDevices(std::vector<std::string>& devices)
+{
+    devices.clear();
+
+    // Create a temporary context
+    ma_context context;
+
+    if (ma_context_init(nullptr, 0, nullptr, &context) != MA_SUCCESS)
+    {
+        CUBOS_ERROR("Failed to initialize audio context for enumeration");
+        return;
+    }
+
```

Comment:
Do not create a temporary context - make this class 'not static' and use the context created in the constructor.